v3/TreeJax/ts/semantic_visitor.ts

```diff
@@ -0,0 +1,510 @@
+/*************************************************************
+ *
+ *  Copyright (c) 2015-2016 The MathJax Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/**
+ * @fileoverview A simple Semantic output visitor.
+ *
+ * @author v.sorge@mathjax.org (Volker Sorge)
+ */
+
+/// <reference path="semantic.d.ts" />
+
+import {Visitor, AbstractVisitor} from './visitor';
+import {TreeNode, Kind, Tag} from './tree_node';
+import {LeafNode, EmptyNode} from './leaf_node';
+import * as nf from './node_factory';
+import sem = require('semantic');
+
+export class SemanticVisitor extends AbstractVisitor {
+
+  private factory: sem.NodeFactory;
+  private result: sem.Tree = sem.emptyTree();
+  private levels: Map<number, sem.Node[]> = new Map<number, sem.Node[]>();
```

Comment:
Rename to something meaningful. childrenStack?
Actually it could be implemented as a stack!